cleancat/sqla.py

```diff
@@ -38,3 +41,28 @@ def fetch_existing(self, pk):
 
     def get_orig_data_from_existing(self, obj):
         return object_as_dict(obj)
+
+
+class SQLAReference(Field):
+    """
+    Represents a reference to an SQLAlchemy object. Expects an ID string as
+    input and returned a cleaned model instance (verifying that it exists
+    first).
+    """
+
+    base_type = str_type
+
+    def __init__(self, model_class, *args, **kwargs):
+        self.model_class = model_class
+        super(SQLAReference, self).__init__(*args, **kwargs)
+
+    def clean(self, value):
+        model_id = super(SQLAReference, self).clean(value)
+        model = self.model_class.query.get(model_id)
```

Comment:
We might want to make this a separate method so subclasses can override it. I think we should have a `Reference` class that takes a `reference_cls` and is subclassable for Mongo and SQLA references. 

Also, If you look at https://github.com/closeio/flask-restutils/blob/master/flask_restutils/resources.py#L36 , we don't use `get()` because it's incompatible with `RandomPKMixin` (it would expect a UUID object rather than the object ID).